### DISCUSSION TITLE
Question on Module Splitting with pthreads

### DISCUSSION THREAD

**Discussion by shinji-TH** [emscripten-core/emscripten#19077] [Q&A] [answered]
From the documentation on Module Splitting [link](https://emscripten.org/docs/optimizing/Module-Splitting.html?highlight=split_module) 

When performing module splitting with multi-threaded apps, the flags "--in-memory" and "-sGLOBAL_BASE=N"  are recommanded to use in order to capture profile information from all the threads. 

However, in the function [do_split_module()](https://github.com/emscripten-core/emscripten/blob/de7cbc56c3ac38b83cc3e8bbc17ac5fbb6b3bbd7/emcc.py#L780), the flag "--in-memory" is not included when the pthread is used when building the instrument version.

Is there any particular reason why the "--in-memory" flag is not included?

**Comment by kripken** [comment, marked as answer]
Looks like this might not be done automatically yet. @tlively ?

**Reply by tlively** [reply]
Right. The usability story for this hasn't been quite fleshed out yet. We could pass `--in-memory` automatically when we are linking a multithreaded application, but we would also need to either generate or validate a reasonable value for `GLOBAL_BASE`. In the long run (i.e. later this year, hopefully), the usability story will be significantly simplified by using multi-memory for instrumentation, avoiding the need to set `GLOBAL_BASE`. Until then, users can re-instrument the original module with the `--in-memory` flag manually.

cc @ashleynh
